lib/components/horizontal_section.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import '../graphql/schema/pages.graphql.dart';
+import '../graphql/queries/calendar_episode_entries.graphql.dart';
+import '../graphql/queries/page.graphql.dart';
+
+import '../providers/todays_calendar_entries.dart';
+import '../helpers/utils.dart';
+import '../models/analytics/sections.dart';
+import 'horizontal_section_episode.dart';
+import 'horizontal_section_show.dart';
+import 'horizontal_slider.dart';
+import 'section_item_click_wrapper.dart';
+
+class _HorizontalSectionSize {
+  final Size imageSize;
+  final double sectionHeight;
```

Comment:
move into the `ItemSectionThumbnailSlider` class as required argments

Reply:
Create factory functions:
ItemSectionThumbnailSlider.fromDefaultSection()
ItemSectionThumbnailSlider.fromPosterSection()